documentation/modules/oauth/con-oauth-authentication-broker.adoc

```diff
@@ -157,6 +158,48 @@ If you want to use Kafka ACL authorization, identify the user by their username
 
+[id='con-oauth-authentication-broker-serviceaccounts-{context}']
+.Configuring fast local JWT token validation with Kubernetes service accounts
+
+To configure the listener for Kubernetes service accounts, the Kubernetes API server must be used as the authorization server.
+
+.Example configuration for fast local JWT token validation using Kubernetes API server as authorization server
+[source,yaml,subs="+quotes,attributes"]
+----
+#...
+- name: external3
+  port: 9094
+  type: loadbalancer
+  tls: true
+  authentication:
+    type: oauth
+    validIssuerUri: https://kubernetes.default.svc.cluster.local # <1>
+    jwksEndpointUri: https://kubernetes.default.svc.cluster.local/openid/v1/jwks # <2>
+    serverBearerTokenLocation: /var/run/secrets/kubernetes.io/serviceaccount/token # <3>
+    checkAccessTokenType: false # <4>
+    includeAcceptHeader: false # <5>
+    tlsTrustedCertificates:
+      - secretName: oauth-server-cert
+        pattern: "*.crt"
+    maxSecondsWithoutReauthentication: 3600
+    customClaimCheck: "@.['kubernetes.io'] && @.['kubernetes.io'].['namespace'] in ['example']" # <6>
+----
+<1> URI of the token issuer used for authentication. Must use FQDN, including the `.cluster.local` extension, which may vary based on the Kubernetes cluster configuration.
+<2> URI of the JWKS certificate endpoint used for local JWT validation. Must use FQDN, including the `.cluster.local` extension, which may vary based on the Kubernetes cluster configuration.
+<3> Location to the access token used by the Kafka broker to authenticate to the Kubernetes API server in order to access the `jwksEndpointUri`.
+<4> Skip the access token type check, as the claim for this is not present in service account tokens.
+<5> Skip sending `Accept` header in HTTP requests to the JWKS endpoint, as the Kubernetes API server does not support it.
+<6> (Optional) Additional constraints that JWT token has to fulfill in order to be accepted, expressed as JsonPath filter query. In this example the service account has to belong to `example` namespace in order to be allowed to authenticate.
+
+The above configuration uses the `sub` claim from the service account JWT token as the user ID. For example, the default service account for pods deployed in  the `example` namespace has the username: `system:serviceaccount:example:default`.
+
+Note that the `tlsTrustedCertificates` is configured to point to a manually created Secret that contains the Kubernetes API server public certificate, which is mounted to the running pods under `/var/run/secrets/kubernetes.io/serviceaccount/ca.crt`.
+The Secret can be created using the following command:
+[source,shell,subs=attributes+]
+----
+kubectl get cm kube-root-ca.crt -o jsonpath="{['data']['ca\.crt']}" > /tmp/ca.crt
+kubectl create secret generic oauth-server-cert --from-file=ca.crt=/tmp/ca.crt
+----
```

Comment:
Within current CRD functionality I didn't find a way to specify this as a path to the file on the disk - basically a passthrough String rather than a Secret, with the expectation that the file is provided to the container by the infrastructure or another operator.

Reply:
@scholzj I don't think you can add a code bolock to the content of the callout. Best I can do is refer to the further section that explains it in more detail.

Reply:
Have you tried something like this? That should work:

```
<6> Trusted certificates to connect to authorization server. This should point to a manually created Secret that contains the Kubernetes API server public certificate, which is mounted to the running pods under `/var/run/secrets/kubernetes.io/serviceaccount/ca.crt`. You can use the following commend to create the Secret:
+
[source,shell,subs=attributes+]
----
kubectl get cm kube-root-ca.crt -o jsonpath="{['data']['ca\.crt']}" > /tmp/ca.crt
kubectl create secret generic oauth-server-cert --from-file=ca.crt=/tmp/ca.crt
----
```

Reply:
![Screenshot 2024-07-25 at 15 01 30](https://github.com/user-attachments/assets/c659a399-76e4-485b-8ab7-117cdf8c53c6)

Reply:
Thanks, I modified it as you suggested. Looks like the plus does the trick.

Reply:
Thanks.